_posts/2019-12-25-github-actions-for-sam.md

```diff
@@ -0,0 +1,368 @@
+
+# What is the Serverless Application Repository?
+
+The Serverless Application Repository (SAR) is a managed repository provided by AWS to host blueprints for serverless applications. If you have a serverless application that would be useful as a reusable component - you can publish it to the SAR and then yourself and others can launch it directly from there. Patterns you create can be shared privately to specific AWS accounts, or you can specify them as public; allowing anyone with an AWS account to discover the pattern and launch it.
+
+# What is Github Actions?
+
+Github Actions is a platform for building workflows that integrate with Github events. The obvious use-case is for CI/CD triggered on various actions directly on a Git repository, but you can also execute logic from actions that originate from PRs, comments, raising issues etc. Workflows and individual actions are defined in YAML and executed via containers or javascript. Custom actions can be stored in other repositories to promote reuse. Github actions is currently in beta, with a plan to become generally available in November 2019.
+
+# Why not use CodePipeline?
+
+The only issue I take with CodePipeline is that it does not support dynamic detection of git references - you must select a particular reference to track (e.g. a specific branch). I find this makes certain workflows difficult to implement without either cluttering the pipeline with branching logic (i.e. do this if this commit is tagged), or introducing additional infrastruture (e.g. CloudWatch Events and Lambda). That being said, CodePipeline does have a lot of things going for it, native integration with AWS IAM being a prime example. Any third-party solution is going to require storing keys to authenticate with AWS which will require more investment in time and effort to maintain an adequate security posture. Like most things in tech there is a trade-off involved.
+
+# My Typical Release Workflow
+
+My typical CI pattern for something contains the following elements;
+
+1. A continuous integration process that builds and tests every push/merge to every branch.
+
+This is to ensure that the branch can be built and tested correctly before pushed to master. If this fails, I want to prevent the code from being merged to master. Once merged, the state of the master branch has changed and therefore I want the tests to be re-executed to ensure the head of master remains in a releasable state.
+
+2. A process that releases an artifact when a tag is pushed to the head of master.
```

Comment:
The numbering when it actually renders doesn't work. Might be better to use dot points instead

![image](https://user-images.githubusercontent.com/7262516/71340890-220e3c80-2593-11ea-83bd-0ec89308e060.png)

Reply:
I've fixed it. Needed to indent the paragraphs under the list in markdown to get the formatting right.